### PR TITLE
Milestone2 friendship v1

### PR DESCRIPTION
**Pull request by Jarfsoft**
**Features added:**
- Friendship feature in the user model
- Unit tests for user, comment, post, like and friendship models

### CODE REVIEW

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Juan Raudales Flores & Álvaro Ruiz Flórez** 
Nice job with your first submission for this project :muscle:  :muscle:
Your project meets most of the requirements :clap: :clap: :clap:  
There are just a few details to improve before you can go ahead:

- [x] timeline should only show comments from user friends
- [ ] User should be able to request friendship to other users
- [ ] Add a screenshot in your README file
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've made laudable progress in this milestone.
However, some changes are required before it can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] I have the error below after installing all dependencies and starting your app:
cms

- [ ] As a result of the bug above, most of your test cases are failing. Kindly fix the bug and ensure that all specs run successfully.



* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, everything works fine, but there are a few more things required. 
### General


- [ ] I shouldn't see post of other users that aren't on my friend's list  ,Lets fix that
- [ ] The comment section needs work on the design

- [ ] Lets work on the friends request feature its still not working


### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by blackpintz — CHANGES_REQUESTED** [review]
# Status: Changes Required 🛑 

### Hello team 👋,

**Great job implementing the requested changes 👏.**

### Highlights

- **Linter checks are configured properly**

### Mandatory requirements.

1).  **A user should also be able to reject a friend request.**
2).  **After a friend request button is sent, the button should switch to a text informing the user that the friend request has been sent. See screenshot below.**

 before a request is sent  | After a request is sent
 ---------------------------| ------------------------
before_request | after_request

3). **Say Jen and Mercy are friends. When Mercy logs in, she should see "Friends" text next to Jen's name. Also, Jen should see a text next to Mercy's name when she logs in.** 

4). **Add tests for your views and helpers.**

### Happy coding 🥂.

**You can reach me on slack (Rose Wanjohi).**

File: spec/views/users/sign_up.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe 'users/sign_up', type: :view do
+  describe 'the signup process', type: :feature do
+    # before :each do
+    #  User.create(name: 'user1', email: 'user1@example.com', password: 'password')
+    # end
```
Comment (acushlakoncept) [inline spec/views/users/sign_up.html.erb_spec.rb:7]: It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

**Review by acushlakoncept — APPROVED** [review]
# Status: PROJECT APPROVED  🎉 🚀

### Congratulations on completing this milestone and taking the time to implement the changes requested by the TSEs. 

 Well done on the specs, consider using [Shoulda-matchers](https://github.com/thoughtbot/shoulda-matchers) for your unit tests, it's a lot easier to perform both unit and association tests. 

Example 
``` 
describe 'validations' do
    it { should validate_presence_of(:name) }
    it { should validate_uniqueness_of(:name).scoped_to(:category_id) }
  end 
```

I will go ahead and mark this as complete, feel free to merge when you are ready. All the best with the next milestone.

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla), if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Pull request merged**